File: 191_number_of_1_bits/191_number_of_1_bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n:
            count += n & 1  # Check the least significant bit
            n >>= 1         # Shift right to check the next bit
        return count
        # Brian Kernighan algorithm
        while n:
            n &= (n - 1)  # Turn off the rightmost set bit
            count += 1
        return count

File: 191_number_of_1_bits/191_number_of_1_bits.js
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function (n) {
  let count = 0;
  // while (n) {
  //     count += n & 1; // Check the least significant bit
  //     // >>>= is known as the unsigned right shift assignment bitwise operator.
  //     n >>>= 1; // Shift right to check the next bit
  // }
  // Brian Kernighan algorithm
  while (n) {
    n &= n - 1;
    count += 1;
  }

  return count;
};
